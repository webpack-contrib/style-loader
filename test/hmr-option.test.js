import webpack from 'webpack';

import compile from './helpers/compiler';
import runTestInJsdom from './helpers/runTestInJsdom';

describe('hmr option', () => {
  const injectTypes = ['styleTag', 'useableStyleTag', 'linkTag'];

  injectTypes.forEach((injectType) => {
    expect.assertions(4);

    it(`not specified without HotModuleReplacementPlugin ("injectType" option is "${injectType}")`, async () => {
      const testId = './hot.js';
      const stats = await compile(testId, { loader: { injectType } });

      runTestInJsdom(stats, (dom) => {
        expect(dom.window.hotApi).not.toBeDefined();
      });

      const bundleSource = stats.compilation.assets['main.bundle.js'].source();

      expect(bundleSource).not.toMatch(/module\.hot\.accept/);

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    it(`not specified with HotModuleReplacementPlugin ("injectType" option is "${injectType}")`, async () => {
      expect.assertions(4);

      const testId = './hot.js';
      const stats = await compile(testId, {
        plugins: [new webpack.HotModuleReplacementPlugin()],
        loader: { injectType },
      });

      runTestInJsdom(stats, (dom) => {
        expect(dom.window.hotApi).toBeDefined();
      });

      const bundleSource = stats.compilation.assets['main.bundle.js'].source();

      expect(bundleSource).toMatch(/module\.hot\.accept/);

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    it(`false without HotModuleReplacementPlugin ("injectType" option is "${injectType}")`, async () => {
      expect.assertions(4);

      const testId = './hot.js';
      const stats = await compile(testId, {
        loader: { injectType, options: { hmr: false } },
      });

      runTestInJsdom(stats, (dom) => {
        expect(dom.window.hotApi).not.toBeDefined();
      });

      const bundleSource = stats.compilation.assets['main.bundle.js'].source();

      expect(bundleSource).not.toMatch(/module\.hot\.accept/);

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    it(`false with HotModuleReplacementPlugin ("injectType" option is "${injectType}")`, async () => {
      expect.assertions(4);

      const testId = './hot.js';
      const stats = await compile(testId, {
        plugins: [new webpack.HotModuleReplacementPlugin()],
        loader: { injectType, options: { hmr: false } },
      });

      runTestInJsdom(stats, (dom) => {
        expect(dom.window.hotApi).toBeDefined();
      });

      const bundleSource = stats.compilation.assets['main.bundle.js'].source();

      expect(bundleSource).not.toMatch(/module\.hot\.accept/);

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    it(`true without HotModuleReplacementPlugin ("injectType" option is "${injectType}")`, async () => {
      expect.assertions(4);

      const testId = './hot.js';
      const stats = await compile(testId, {
        loader: { injectType, options: { hmr: true } },
      });

      runTestInJsdom(stats, (dom) => {
        expect(dom.window.hotApi).not.toBeDefined();
      });

      const bundleSource = stats.compilation.assets['main.bundle.js'].source();

      expect(bundleSource).not.toMatch(/module\.hot\.accept/);

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    it(`true with HotModuleReplacementPlugin ("injectType" option is "${injectType}")`, async () => {
      expect.assertions(4);

      const testId = './hot.js';
      const stats = await compile(testId, {
        plugins: [new webpack.HotModuleReplacementPlugin()],
        loader: { injectType, options: { hmr: true } },
      });

      runTestInJsdom(stats, (dom) => {
        expect(dom.window.hotApi).toBeDefined();
      });

      const bundleSource = stats.compilation.assets['main.bundle.js'].source();

      expect(bundleSource).toMatch(/module\.hot\.accept/);

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });
  });
});
