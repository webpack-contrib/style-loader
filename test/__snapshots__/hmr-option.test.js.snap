// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hmr option false with HotModuleReplacementPlugin ("injectType" option is "linkTag"): errors 1`] = `Array []`;

exports[`hmr option false with HotModuleReplacementPlugin ("injectType" option is "linkTag"): warnings 1`] = `Array []`;

exports[`hmr option false with HotModuleReplacementPlugin ("injectType" option is "styleTag"): errors 1`] = `Array []`;

exports[`hmr option false with HotModuleReplacementPlugin ("injectType" option is "styleTag"): warnings 1`] = `Array []`;

exports[`hmr option false with HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): errors 1`] = `Array []`;

exports[`hmr option false with HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): warnings 1`] = `Array []`;

exports[`hmr option false without HotModuleReplacementPlugin ("injectType" option is "linkTag"): errors 1`] = `Array []`;

exports[`hmr option false without HotModuleReplacementPlugin ("injectType" option is "linkTag"): warnings 1`] = `Array []`;

exports[`hmr option false without HotModuleReplacementPlugin ("injectType" option is "styleTag"): errors 1`] = `Array []`;

exports[`hmr option false without HotModuleReplacementPlugin ("injectType" option is "styleTag"): warnings 1`] = `Array []`;

exports[`hmr option false without HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): errors 1`] = `Array []`;

exports[`hmr option false without HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): warnings 1`] = `Array []`;

exports[`hmr option not specified with HotModuleReplacementPlugin ("injectType" option is "linkTag"): errors 1`] = `Array []`;

exports[`hmr option not specified with HotModuleReplacementPlugin ("injectType" option is "linkTag"): warnings 1`] = `Array []`;

exports[`hmr option not specified with HotModuleReplacementPlugin ("injectType" option is "styleTag"): errors 1`] = `Array []`;

exports[`hmr option not specified with HotModuleReplacementPlugin ("injectType" option is "styleTag"): warnings 1`] = `Array []`;

exports[`hmr option not specified with HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): errors 1`] = `Array []`;

exports[`hmr option not specified with HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): warnings 1`] = `Array []`;

exports[`hmr option not specified without HotModuleReplacementPlugin ("injectType" option is "linkTag"): errors 1`] = `Array []`;

exports[`hmr option not specified without HotModuleReplacementPlugin ("injectType" option is "linkTag"): warnings 1`] = `Array []`;

exports[`hmr option not specified without HotModuleReplacementPlugin ("injectType" option is "styleTag"): errors 1`] = `Array []`;

exports[`hmr option not specified without HotModuleReplacementPlugin ("injectType" option is "styleTag"): warnings 1`] = `Array []`;

exports[`hmr option not specified without HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): errors 1`] = `Array []`;

exports[`hmr option not specified without HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): warnings 1`] = `Array []`;

exports[`hmr option true with HotModuleReplacementPlugin ("injectType" option is "linkTag"): errors 1`] = `Array []`;

exports[`hmr option true with HotModuleReplacementPlugin ("injectType" option is "linkTag"): warnings 1`] = `Array []`;

exports[`hmr option true with HotModuleReplacementPlugin ("injectType" option is "styleTag"): errors 1`] = `Array []`;

exports[`hmr option true with HotModuleReplacementPlugin ("injectType" option is "styleTag"): warnings 1`] = `Array []`;

exports[`hmr option true with HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): errors 1`] = `Array []`;

exports[`hmr option true with HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): warnings 1`] = `Array []`;

exports[`hmr option true without HotModuleReplacementPlugin ("injectType" option is "linkTag"): errors 1`] = `Array []`;

exports[`hmr option true without HotModuleReplacementPlugin ("injectType" option is "linkTag"): warnings 1`] = `Array []`;

exports[`hmr option true without HotModuleReplacementPlugin ("injectType" option is "styleTag"): errors 1`] = `Array []`;

exports[`hmr option true without HotModuleReplacementPlugin ("injectType" option is "styleTag"): warnings 1`] = `Array []`;

exports[`hmr option true without HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): errors 1`] = `Array []`;

exports[`hmr option true without HotModuleReplacementPlugin ("injectType" option is "useableStyleTag"): warnings 1`] = `Array []`;
